### PR TITLE
Use long everywhere a oid is used

### PR DESCRIPTION
**Pull request by zeitlinger**
- Use long everywhere a oid is used - adapted from https://github.com/zalando-stups/java-sproc-wrapper/pull/59
- Upgrade the postgres driver
- fix tests

has been tested in ZMON - where the problem actually existed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,10 +59,10 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
-        <spring.version>5.2.16.RELEASE</spring.version>
-        <postgresql.version>42.2.14</postgresql.version>
+        <maven.compiler.source>8</maven.compiler.source>
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply pom.xml:62]: ZMON controller is still using Java 8. I could only see a single use of Java 11, so I figured it's easier to backport this lib to Java 8 than to port ZMON controller to Java 11.

[the remark this answers is not in the document]
Comment (vadeg) [reply pom.xml:62]: It is not only ZMON controller. Also from a security perspective it is not a good idea to backport.

[the remark this answers is not in the document]
Comment (zeitlinger) [reply pom.xml:62]: this is a library that sets the minimum Java version - the actual Java version is defined by the application.
So while I agree that ZMON controller should probably be upgraded to a more recent Java version - we don't want to make this a pre-requisite to upgrade to Postgres 14.

**Comment by vadeg** [comment]
Need to update spring version:
```
Error:  One or more dependencies were identified with vulnerabilities: 
Error:  
Error:  spring-core-5.3.23.jar: CVE-2016-1000027
```

File: src/test/java/org/zalando/typemapper/postgres/PgSerializerTest.java [hunk 2]
```diff
@@ -83,19 +79,23 @@ public static Collection<Object[]> generateData() throws SQLException {
                                 ARRAY("first element", "second \"quoted\" element").asJdbcArray("text"),
                                 "{\"first element\",\"second \\\"quoted\\\" element\"}"
                         },
-                        {ROW(1, 2), "(1,2)"},
-                        {ROW("a", "b{x}"), "(a,b{x})"},
-                        {ROW("a", "b(x)"), "(a,\"b(x)\")"},
-                        {ROW(1, 2, ARRAY("a", "b")), "(1,2,\"{a,b}\")"},
-                        {ROW("a", "b", new int[]{1, 2, 3, 4}), "(a,b,\"{1,2,3,4}\")"},
-                        {ROW("a", null, ARRAY(ROW(1), ROW(2), null)), "(a,,\"{(1),(2),NULL}\")"},
+                        {row(1, 2), "(1,2)"},
```
Comment (vadeg) [inline src/test/java/org/zalando/typemapper/postgres/PgSerializerTest.java:82]: I don't think this is the correct fix. You've implemented separate version of `row` while tests and the library uses `PgRow.row` which is used in different parts of the library including mapper for procedure parameters (`OtherStoredProcedureParameter`). This fix makes test work, but in production the behaviour could be different. 

I suggest to apply a fix to `PgRow` and use the same `row` implementation for tests and the library itself.

**Comment by zeitlinger** [comment]
> Need to update spring version:
> 
> ```
> Error:  One or more dependencies were identified with vulnerabilities: 
> Error:  
> Error:  spring-core-5.3.23.jar: CVE-2016-1000027
> ```

It is already at the latest version - see https://mvnrepository.com/artifact/org.springframework/spring-core

Should we add an exception? And if so, how?

File: src/test/java/org/zalando/typemapper/postgres/PgSerializerTest.java (hunk 2 above)
Comment (zeitlinger) [reply src/test/java/org/zalando/typemapper/postgres/PgSerializerTest.java]: done

**Review by vadeg — APPROVED** [review]

**Comment by vadeg** [comment]
👍

**Pull request merged**